vignettes/join-keys.Rmd

```diff
@@ -8,39 +8,98 @@ vignette: >
+- `join_key(...)` specifies the relation between two datasets:
+  - `dataset_1`, `dataset_2` - name of each datasets (if `dataset_2` is the same as `dataset_1` then it creates a primary key)
+  - `key` - (optionally) named vector of column names
+
+Note that it is assumed that join keys are symmetric, i.e. `join_key("x", "y", "x_col" = "y_col")` will enable merge
+from "x" to "y" and vice versa.
+
+```{r, eval=FALSE}
+jk <- join_keys(
+  join_key("ds1", keys = "id1"), # ds1: [id1]
+  join_key("ds2", keys = c("id1", "id2")), # ds2: [id1, id2]
+  join_key("ds3", keys = c("id1", "id3")), # ds3: [id1, id3]
+  join_key("ds1", "ds2", keys = "id1"), # ds1 <--> ds2
+  join_key("ds1", "ds3", keys = "id1"), # ds1 <--> ds3
+  join_key("ds4", "ds5", keys = c("id4" = "id5")) # ds4 <--> ds5
+)
+
+# The parent of ds2 and ds3 is ds1
+# converts relationship to child-parent
+#  ds1 <--> ds2 becomes ds1 <-- ds2
+#  ds1 <--> ds3 becomes ds1 <-- ds3
+parents(jk) <- list(ds2 = "ds1", ds3 = "ds1")
+
+jk
+```
+
+| Output of `print(jk)`              | Output annotation                    |
+| ---------------------------------- |:----------------------------------------:|
+| `## A join_keys object containing foreign keys between 3 datasets:` |  **Title** |
+| `## ds1: [id1]`                |  **Primary keys**<br />_\<dataset name\>_: [_\<primary_keys\>_]                                          | 
+| `##   <-- ds2: [id1]`          | **Foreign keys**<br /> _(arrow `<--` denotes `ds1` is the parent of `ds2`)_                    | 
+| `##   <-- ds3: [id1]`          |                                                                               | 
+| `## ds2: [id1, id2]`           |                                                                               | 
+| `##   --> ds1: [id1]`          | arrow `-->` denotes `ds2` is a child of `ds1`                                                                              | 
+| `##   --* (implicit via parent with): ds3` |  **Implicit relationship between `ds2` & `ds3`**<br />_(given that they share common keys with same parent)_    |
+| `## ds3: [id1, id3]`           |                                                                               | 
+| `##   --> ds1: [id1]`          |                                                                               | 
+| `##   --* (implicit via parent with): ds2` |                                                                               | 
+| `## ds4: [no primary keys]`    |                                                                               | 
+| `##   <-> ds5: [id5]`          |  **Foreign keys** <br />_(arrow `<->` denotes no parent definition between datasets)_      | 
+| `## ds5: [no primary keys]`    |                                                                               | 
+| `##   <-> ds4: [id4] `         |                                                                               | 
+
```

Comment:
I love the idea of a table like this. The only improvement would be row borders so it's easy to read.
<img width="856" alt="Screenshot 2023-11-28 at 12 58 22 PM" src="https://github.com/insightsengineering/teal.data/assets/49812166/e2e7b2fe-5689-401c-990b-52f902d4988b">

Reply:
I think the template being used on the site will add those rows

From [another vignete](https://insightsengineering.github.io/teal.data/latest-tag/articles/using-delayed-data-basic.html)

![image](https://github.com/insightsengineering/teal.data/assets/211358/bee46ad8-3413-4c75-8de3-b67c3825db25)